external/storm-jms/src/main/java/org/apache/storm/jms/spout/JmsSpout.java

```diff
@@ -339,26 +339,26 @@ public void nextTuple() {
      */
     @Override
     public void ack(Object msgId) {
-
         Message msg = this.pendingMessages.remove(msgId);
-        JmsMessageID oldest = this.toCommit.first();
-        if (msgId.equals(oldest)) {
-            if (msg != null) {
-                try {
-                    LOG.debug("Committing...");
-                    msg.acknowledge();
-                    LOG.debug("JMS Message acked: " + msgId);
-                    this.toCommit.remove(msgId);
-                } catch (JMSException e) {
-                    LOG.warn("Error acknowldging JMS message: " + msgId, e);
+        if (!toCommit.isEmpty()) {
```

Comment:
Could we leave the log message (at least DEBUG) so that we can see which messages are ignored while acking due to previous failure?

Reply:
Might be better to leave the log message which messages are dropped in `fail()` too.